src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
+            {
+                return ImmutableArray<Document>.Empty;
+            }
 
-        /// <summary>
-        /// Gets the text after the last parsed span available. This is after
-        /// file information if it is available, otherwise after the argument information.
-        /// </summary>
-        public string GetTrailingText()
-        {
-            var lastSpan = FileSpan == default
-                ? ArgsSpan
-                : FileSpan;
+            var fileName = Root.FileInformationExpression.Path.ToString();
+            var lineString = Root.FileInformationExpression.Line.ToString();
+            RoslynDebug.AssertNotNull(lineString);
+            lineNumber = int.Parse(lineString);
 
-            if (lastSpan.End + 1 == OriginalText.Length)
+            var documentName = Path.GetFileName(fileName);
+            var potentialMatches = new HashSet<Document>();
+
+            foreach (var project in solution.Projects)
             {
-                return string.Empty;
+                foreach (var document in project.Documents)
+                {
+                    if (document.FilePath == fileName)
+                    {
+                        return ImmutableArray.Create(document);
+                    }
+
+                    else if (document.Name == documentName)
+                    {
+                        potentialMatches.Add(document);
+                    }
+                }
             }
 
-            return OriginalText[(lastSpan.End + 1)..];
+            return potentialMatches.ToImmutableArray();
         }
 
-        /// <summary>
-        /// If the frame has file information, gets the text between the method and file information.
-        /// ex: at ConsoleApp4.MyClass.M[T](T t) in [|C:\repos\Test\MyClass.cs:line 7|]
-        /// </summary>
-        public string? GetFileText()
+        private static bool MatchParameters(ImmutableArray<IParameterSymbol> parameters, StackFrameParameterList stackFrameParameters)
         {
-            if (FileSpan == default)
+            if (parameters.Length != stackFrameParameters.Parameters.Length)
             {
-                return null;
+                return false;
             }
 
-            return OriginalText[FileSpan.Start..FileSpan.End];
-        }
-
-        /// <summary>
-        /// If the frame has file information, gets the text between the method and file information.
-        /// ex: at ConsoleApp4.MyClass.M[T](T t)[| in |]C:\repos\Test\MyClass.cs:line 7
-        /// </summary>
-        public string? GetTextBetweenTypeAndFile()
-        {
-            if (FileSpan == default)
+            for (var i = 0; i < stackFrameParameters.Parameters.Length; i++)
             {
-                return null;
+                var stackFrameParameter = stackFrameParameters.Parameters[i];
+                var paramSymbol = parameters[i];
+
+                if (paramSymbol.Name != stackFrameParameter.Identifier.ToString(skipTrivia: true))
```

Comment:
note: i would be amenable to EmbeddedSyntaxToken getting a ValueText property like normal SyntaxToken has.  that would match this better.